genesis/utils/array_class.py

```diff
@@ -20,6 +20,10 @@
 DATA_ORIENTED = partial(dataclasses.dataclass, frozen=True) if gs.use_ndarray else ti.data_oriented
 
 
+def _s(shape, is_on):
+    return shape if is_on else (1,)
```

Comment:
Besides, if you want to avoid memory allocation for unused variables, you should rather use shape `()` instead of `(1,)`.

Reply:
done